static/js/upload.js

```diff
@@ -13,187 +17,223 @@ exports.feature_check = function (upload_button) {
+        }
+        exports.upload_files(uppy, config, files);
+    });
 
-    const uploadFinished = function (i, file, response) {
-        if (response.uri === undefined) {
+    uppy.on('upload-success', (file, response) => {
+        const uri = response.body.uri;
+        if (uri === undefined) {
             return;
         }
-        const split_uri = response.uri.split("/");
+        const split_uri = uri.split("/");
         const filename = split_uri[split_uri.length - 1];
-        // Urgh, yet another hack to make sure we're "composing"
-        // when text gets added into the composebox.
-        if (config.mode === 'compose' && !compose_state.composing()) {
+        if (!compose_state.composing()) {
             compose_actions.start('stream');
-        } else if (config.mode === 'edit' && document.activeElement !== textarea) {
-            // If we are editing, focus on the edit message box
-            textarea.focus();
-        }
-
-        const uri = make_upload_absolute(response.uri);
-
-        if (i === -1) {
-            // This is a paste, so there's no filename. Show the image directly
-            const pasted_image_uri = "[pasted image](" + uri + ")";
-            compose_ui.replace_syntax("[Uploading " + file.name + "…]()", pasted_image_uri, textarea);
-        } else {
-            // This is a dropped file, so make the filename a link to the image
-            const filename_uri = "[" + filename + "](" + uri + ")";
-            compose_ui.replace_syntax("[Uploading " + file.name + "…]()", filename_uri, textarea);
         }
+        const absolute_uri = upload.make_upload_absolute(uri);
+        const filename_uri = "[" + filename + "](" + absolute_uri + ")";
+        compose_ui.replace_syntax("[Uploading " + file.name + "…]()", filename_uri, exports.get_item("textarea", config));
         compose_ui.autosize_textarea();
+    });
 
+    uppy.on('complete', () => {
         setTimeout(function () {
-            $("#" + upload_bar  + '-' + file.trackingId).parent().remove();
-            if ($('div.progress.active').length === 0) {
-                hide_upload_status(file);
-            }
+            uppy.cancelAll();
+            exports.hide_upload_status(config);
         }, 500);
-
-        // In order to upload the same file twice in a row, we need to clear out
-        // the file input element, so that the next time we use the file dialog,
-        // an actual change event is fired. IE doesn't allow .val('') so we
-        // need to clone it. (Taken from the jQuery form plugin)
-        if (/MSIE/.test(navigator.userAgent)) {
-            $('#' + file_input).replaceWith($('#' + file_input).clone(true));
-        } else {
-            $('#' + file_input).val('');
+    });
+
+    uppy.on('info-visible', () => {
+        const info = uppy.getState().info;
+        // Ideally we should handle "Upload Error" case here, but info.message
+        // don't contain response.body.msg received from the server.
```

Comment:
Can you extend this comment to explain when this `info-visible` case happens? I'm not sure how common this will be.

Reply:
Done. 

I just realized that I dont hide the "disconnected to Internet error" when Internet comes back. Uppy generates an event when the Internet comes back and this can be caputured by `uppy.on('info-visible')`. Let me take care of that.